thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java

```diff
@@ -308,6 +314,33 @@ public DetectionPipelineResult run() throws Exception {
         predictionResults, calculateEvaluationMetrics(predictionResults)).setDiagnostics(diagnostics);
   }
 
+  private DatasetConfigDTO retrieveDatasetConfig(List<MetricEntity> nestedMetrics) {
+    // All nested metrics after dimension explore belong to the same dataset
+    long metricId = nestedMetrics.get(0).getId();
+    MetricConfigDTO metricConfigDTO = this.provider.fetchMetrics(Collections.singletonList(metricId)).get(metricId);
+
+    return this.provider.fetchDatasets(Collections.singletonList(metricConfigDTO.getDataset()))
+        .get(metricConfigDTO.getDataset());
+  }
+
+  private void warmUpAnomaliesCache(List<MetricEntity> nestedMetrics) {
+    DatasetConfigDTO dataset = retrieveDatasetConfig(nestedMetrics);
+    long cachingPeriodLookback = config.getProperties().containsKey(PROP_CACHE_PERIOD_LOOKBACK) ?
+        MapUtils.getLong(config.getProperties(), PROP_CACHE_PERIOD_LOOKBACK) : ThirdEyeUtils
+        .getCachingPeriodLookback(dataset.bucketTimeGranularity());
+
+    long paddingBuffer = TimeUnit.DAYS.toMillis(1); // 100000
+    AnomalySlice anomalySlice = new AnomalySlice()
```

Comment:
Why don't we need to include filters and detection component names when initializing this anomaly slice?